etcdmain/config.go

```diff
@@ -230,6 +234,10 @@ func NewConfig() *config {
 	fs.Var(&flags.DeprecatedFlag{Name: "peers"}, "peers", "DEPRECATED: Use -initial-cluster instead.")
 	fs.Var(&flags.DeprecatedFlag{Name: "peers-file"}, "peers-file", "DEPRECATED: Use -initial-cluster instead.")
 
+	// profiler
+	fs.BoolVar(&cfg.enablePprof, "enable-pprof", false, "Enable pprof profiler")
+	fs.StringVar(&cfg.pprofAddr, "pprof-addr", "localhost:6060", "address for pprof profiler")
```

Comment:
How about adding a handler to client url instead at https://github.com/coreos/etcd/blob/master/etcdserver/etcdhttp/client.go#L103?